test/summary.cc

```diff
@@ -95,13 +95,23 @@ TEST(Summary, test_time_summary) {
     for (const auto& leg : route.legs()) {
       // accumulate the maneuvers for a leg
       double accumulated_time = 0;
+      double accumulated_transition_time = 0;
       for (const auto& maneuver : leg.maneuver()) {
         accumulated_time += maneuver.time();
+        // check the transition times should be non-zero and less than equal to the maneuver time
+        double transition_time = 0;
+        for (auto n = maneuver.begin_path_index(); n < maneuver.end_path_index(); ++n) {
+          transition_time += trip_leg->node(n).transition_time();
+        }
+        EXPECT_LE(transition_time, maneuver.time());
+        accumulated_transition_time += transition_time;
       }
       // make sure the end of the trip path is the same as the legs
       EXPECT_EQ(trip_leg->node().rbegin()->elapsed_time(), leg.summary().time());
       // make sure the maneuvers add up to the leg as well
       EXPECT_EQ(accumulated_time, leg.summary().time());
+      // we should have had some transition costs along the way
+      EXPECT_GT(accumulated_transition_time, 0);
```

Comment:
Should the test also check that the outer duration equals the inner durations plus the transition costs?

Reply:
just added that ;)